azurelinuxagent/common/protocol/wire.py

```diff
@@ -969,7 +970,13 @@ def get_certs(self):
     def get_extensions_goal_state(self):
         if self._extensions_goal_state is None:
             raise ProtocolError("Trying to fetch ExtensioalState before initialization!")
-        return self._extensions_goal_state
+
+        try:
+            ExtensionsGoalState.compare(self._goal_state.extensions_config,  self._extensions_goal_state)
```

Comment:
Any particular reason why comparing here? I thought we were only comparing when we get a new GS

Reply:
oh, sorry, i forgot to remove this debugging code (there was an issue reporting status). I'll remove this and the comments below, thanks for pointing it out.